client/tests/unit/CosmologyForm.spec.js

```diff
@@ -0,0 +1,108 @@
+/* eslint-disable no-await-in-loop, no-restricted-syntax, no-continue */
+
+import { mount, createLocalVue } from '@vue/test-utils';
+import CosmologyForm from '@/components/forms/CosmologyForm';
+import BACKEND_CONSTANTS from '@/constants/backend_constants';
+import INITIAL_STATE from '@/constants/initial_state';
+
+describe('Mounted CosmologyForm', () => {
+  const localVue = createLocalVue();
+  const wrapper = mount(CosmologyForm, {
+    propsData: {
+      title: 'Cosmology',
+      init: INITIAL_STATE.cosmology,
+    },
+    localVue,
+  });
+
+  const options = Object.keys(BACKEND_CONSTANTS.Cosmo_params);
+
+  test('has correct default model', async () => {
+    await wrapper.vm.$nextTick();
+    expect(wrapper.vm.model.cosmo_model).toBe('Planck15');
+  });
+
+  test('changes model parameters when model is changed', async () => {
+    for (const option of options) {
+      if (wrapper.vm.model.cosmo_model === option) continue;
+      const oldParams = wrapper.vm.model.cosmo_params;
+      wrapper.vm.$data.model.cosmo_model = option;
+      await localVue.nextTick();
+      await localVue.nextTick();
+      const newParams = wrapper.vm.model.cosmo_params;
+      expect(oldParams).not.toBe(newParams);
+    }
+  });
```

Comment:
Cosmo is weird because core parameters (things outside cosmo_params) change with model changes, too. Should have tests to reflect this.

Reply:
What core parameters?

If core parameters change, model changes, therefore. There's no need to test very highly coupled parts, because we may want to change that, but not break the tests. 
Testing that this component returns the right output given the inputs should be sufficient.

Reply:
It's testing our logic within the watch function not whether the watch triggers. I agree that we don't need to test Vue features but we do need to test whether the code we wrote using Vue features produces expected results.

In the same vein, it's not testing whether or not about testing whether 'updated' triggers, its testing whether or not the correct thing happens _when_ updated is triggered.

Reply:
Done